spec/integration/primitives_spec.rb

```diff
@@ -0,0 +1,76 @@
+require 'spec_helper'
+require 'time'
+
+describe "chef-provisioning-docker" do
+  extend DockerSupport
+
+  when_the_chef_12_server "exists", organization: "foo", server_scope: :context do
+    with_docker "integration tests" do
+
+      ubuntu_options = {
+          :base_image => {
+            :name => 'ubuntu',
+            :repository => 'ubuntu',
+            :tag => '14.04'
+          },
+        }
+
+      let(:iso_date) { Time.now.iso8601.gsub(':', '') }
+      docker_driver = driver
+
+      context "machine_image resource" do
+
+        let(:spec_image_tag) { "docker_image_spec_#{iso_date}" }
+
+        after(:each) {
+          image = docker_driver.find_image("chef", spec_image_tag)
+          image.delete(force: true) unless image.nil?
+        }
+
+        it ":create succeeds" do
+          tag = spec_image_tag
+
+          expect_converge {
+            with_driver 'docker'
```

Comment:
We should update `with_docker` so that it sets the driver to be docker and we don't need this in every test